src/controller.coffee

```diff
@@ -3118,28 +3136,38 @@ define ['droplet-helper',
         @highlightCanvas.style.opacity =
         @cursorCanvas.style.opacity = 0
 
-      setTimeout (=>
-        @dropletElement.style.transition =
-          @paletteWrapper.style.transition = "left #{translateTime}ms"
+      paletteDisappearingWithMelt = @paletteEnabled and not @showPaletteInTextMode
 
-        @dropletElement.style.left = '0px'
-        @paletteWrapper.style.left = "#{-@paletteWrapper.offsetWidth}px"
-      ), fadeTime
+      if paletteDisappearingWithMelt
+        # Move the palette header into the background
+        @paletteHeader.style.zIndex = 0
+
+        setTimeout (=>
+          @dropletElement.style.transition =
+            @paletteWrapper.style.transition = "left #{translateTime}ms"
+
+          @dropletElement.style.left = '0px'
+          @paletteWrapper.style.left = "#{-@paletteWrapper.offsetWidth}px"
+        ), fadeTime
 
       setTimeout (=>
         # Translate the ICE editor div out of frame.
         @dropletElement.style.transition =
           @paletteWrapper.style.transition = ''
 
-        @dropletElement.style.top = '-9999px'
-        @dropletElement.style.left = '-9999px'
+        # Translate the ACE editor div into frame.
+        @aceElement.style.top = '0px'
+        if @showPaletteInTextMode and @paletteEnabled
+          @aceElement.style.left = @paletteWrapper.style.width
```

Comment:
Works better if we position the aceElement based on paletteWrapper.offfsetWidth.  (In case there's some CSS such as maxWidth that overrides the effective width of the paletteWrapper, as happens in pencil code).  Tweaking this on merge.